File: code/exceptions.kt
try {
    // some code
}
catch (e: SomeException) {
    if (e.SomeCode == 404) {
        // handle SomeException when SomeCode is 404
    } else {
        // handle SomeException
    }
}
catch (e: Exception) {
    // handle rest of the Exceptions
}
finally {
    // optional finally block
}

File: code/exceptions.cs
try 
{
    // Some code
}
catch (SomeException e) when (e.SomeCode == 404) 
{
    // Handle SomeException only when SomeCode is 404
}
catch (SomeException e) 
{
    // Handle SomeException
}
catch (Exception e) 
{
    // Handle rest of the Exceptions 
}
finally 
{
    // Optional finally block
}